Week1/homework/app.js

```diff
@@ -1,11 +1,128 @@
-    'harry_potter_chamber_secrets'
+
+
+  const myBooksName = [
+"milk_and_vine" , "the_little_drummer_girl" , "of_blood_and_bone" , "year_one" ,"come_sundown","vine_and_honey","this_is_marketing","ottolenghi_simple","birthday_girl","the_rebels_revenge"
   ];
 
 
-  // Replace with your own code
-  console.log(bookTitles);
+const details = {
+
+  "milk_and_vine": {
+    title:"Milk and Vine ",
+author:"Adam Gasiewski ",
+language:"English",
+
+
+} ,
+
+
+"the_little_drummer_girl" : {
+  title:"The Little Drummer Girl",
+author:" John Le Carré ",
+language:"English",
+} ,
+
+"of_blood_and_bone": {
+  title:"Of Blood and Bone",
+author:" Nora Roberts ",
+language:"English",
+} ,
+
+"year_one": {
+  title:"Year One",
+  author:" Nora Roberts ",
+  language:"English",
+  } ,
+
+"come_sundown": {
+  title:"Come Sundown",
+author:"  Nora Roberts ",
+language:"English",
+} ,
+
+"vine_and_honey": {
+  title:"Vine and Honey ",
+author:" Ben Mark ",
+language:"English",
+} ,
+
+"this_is_marketing": {
+  title:"This is Marketing ",
+author:" Seth Godin ",
+language:"English",
+} ,
+
+"ottolenghi_simple": {
+  title:"Ottolenghi SIMPLE",
+author:"  Yotam Ottolenghi ",
+language:"English",
+} ,
+
+"birthday_girl": {
+  title:"Birthday Girl",
+author:"Birthday Girl",
+language:"English",
+} ,
+
+"the_rebels_revenge": {
+title:"The Rebel's Revenge",
+author:"  Scott Mariani ",
+language:"English",
 }
+};
+
+const myBookImages = {
+  milk_and_vine : "images/book1.jpg" , 
+  the_little_drummer_girl :"images/book2.jpg" ,
+   of_blood_and_bone :"images/book3.jpg" ,
+    year_one :"images/book4.jpg",
+    come_sundown:"images/book5.jpg",
+    vine_and_honey:"images/book6.jpg",
+    this_is_marketing:"images/book7.jpg",
+    ottolenghi_simple:"images/book8.jpg",
+    birthday_girl:"images/book9.jpg",
+    the_rebels_revenge:"images/book10.jpg",   
+  };
+
+
+function booksList(){
+  const div = document.getElementById("mydiv");
+  let myHeader = document.createElement('h1');
+  div.appendChild(myHeader);
+  let myUl = document.createElement('ul');
+  div.appendChild(myUl);
+
+  let theHeaderContent = document.createTextNode ('My library ');
+  myHeader.appendChild(theHeaderContent);
+
+for (let i=0; i<myBooksName.length; i++){
+let myLi=document.createElement('li');
+//myLi.textContent = myBooksName[i];
+myUl.appendChild(myLi);
+
+let bookName= document.createTextNode(Object.values(details)[i].title);
+let bookAuthor= document.createTextNode( "It is written by " + Object.values(details)[i].author);
+let bookLanguage= document.createTextNode("It is  written in : " + Object.values(details)[i].language);
+
+let myH1 = document.createElement("h2");
+let myH3 = document.createElement("h3");
+let myH4 = document.createElement("h3");
+
+myH1.appendChild(bookName);
+myH3.appendChild(bookAuthor);
+myH4.appendChild(bookLanguage);
+myLi.innerHTML =  myH1.outerHTML + myH3.outerHTML + myH4.outerHTML ;
+
+let bookCovers = document.createElement("img");
```

Comment:
This is very good and it works! 

But just for the sake of following the homework assignment, this part specifically : 
> Loop over these entries (hint: Object.keys(objectName) gives you an array containing the keys). Then write a function which places an image at the corresponding li element. Remember that objects are not ordered, so you cannot guarantee that the first key is the first li element. (Hint: you could give each li item an id tag by modifying the function you made before.)

After the first _for_ loop, the assignment said to do another loop through the keys of `myBookImages` and add the image to the corresponding _li_. In order to do this, you would need to add an _id_ attribute to each _li_, probably the key in `details` object itself (which correspond with keys in `myBookImages`). Then, the second loop would go something like this:

```javascript
let imageKeys = Object.keys(myBookImages);
for(i = 0; i < imageKeys.length; i++){
   let theLi = document.getElementById(imageKeys[i]);
   if (theLi){
      // the code you already wrote here
      let bookCovers = document.createElement("img");
      bookCovers.setAttribute("src", myBookImages[imageKeys[i]]);
      bookCovers.setAttribute("alt", details[imageKeys[i]].title);
      theLi.appendChild(bookCovers);
   }
}
```
and avoiding again the _Object.values(...)_ construct